libsolidity/formal/Predicate.cpp

```diff
@@ -298,6 +295,64 @@ optional<string> Predicate::expressionToString(smtutil::Expression const& _expr,
 		solAssert(_expr.arguments.empty(), "");
 		return _expr.name;
 	}
+	if (smt::isArray(*_type))
+	{
+		auto const& arrayType = dynamic_cast<ArrayType const&>(*_type);
+		solAssert(_expr.name == "tuple_constructor", "");
+		auto const& tupleSort = dynamic_cast<TupleSort const&>(*_expr.sort);
+		solAssert(tupleSort.components.size() == 2, "");
+		// The solver needs to build a concrete counterexample of size at least length,
+		// so we should also throw if the length does not fit an integer.
+		unsigned length = stoul(_expr.arguments.at(1).name);
+		vector<string> array(length);
```

Comment:
Maybe like this as it is used only here:
```suggestion
		vector<string> array(stoul(_expr.arguments.at(1).name));
```